CMakeLists.txt

```diff
@@ -87,15 +81,22 @@ if(OTF2XX_WITH_MPI)
         INTERFACE
             OTF2XX_HAS_MPI
     )
+    find_package(Boost REQUIRED COMPONENTS mpi)
+    find_package(MPI REQUIRED COMPONENTS CXX)
+    if(NOT TARGET MPI::MPI_CXX) # For CMake < 3.9
```

Comment:
This is pure madness. I refuse to merge code duplicates based on some random CMake version decision.
Either a) without these checks, because they are not required, or b) with a CMake version, which provides the targets.

Reply:
Disclaimer: I want to use OTF2xx in a project where the requirement is CMake 3.8 and raising that is/might be problematic.   
CMake 3.8 was chosen (e.g.) to support proper C++ standards which is very important (the other solutions have various shortcomings not easily worked around)

What is wrong with this? When sticking to CMake 3.8 this **is** required. The alternative would be to add a bunch of `target_include_directories`, `target_compile_definitions` and `target_link_libraries` to the code. But this is forward compatible so it allows improvements from using newer CMake versions without any changes.

So what do you mean by "code duplicates"? Except the outer if-endif the alternative would be the same (and IMO even more verbose)

Reply:
The code duplicate is to define MPI targets, just because someone decided CMake newer than some random number is the work of the devils. I don't like that and it either stays like it is, because It is not required. The contract is: "If you want to have MPI (by setting OTF2XX_WITH_MPI to ON), then you have to provide the dependencies yourself.".

Or it will be merge with MPI targets from a decently new CMake version.

Reply:
We are talking about (Boost)MPI. I see no scenario, where you can use MPI "self-contained" in a library, especially when the interface demands a valid boost communicator being passed to it. You therefore have to setup a suitable communicator and provide it to otf2xx and therefore have to do the dependency checks yourself anyway.

Reply:
1. Being able to do `cmake .. -DOTF2XX_WITH_MPI=ON` could be used in CI to check that all MPI code is valid too
2. Enabling MPI does not mean that it must be used. Correct me if I'm wrong, but you can use all the serial stuff from otf2xx even with `-DOTF2XX_WITH_MPI=ON`.
3. You could add an MPI test to otf2xx tests where the required `MPI_Init` (or whatever) is called and the MPI code is tested in the same build as the non-MPI tests.

This is what I mean by "self-contained".

Reply:
In principle, I'm not opposed to add and expose the MPI dependency. I'm opposed against duplicating code, which exist in CMake 3.9 onwards. I won't merge it this they. Period.

If your project insist on using CMake 3.8, fine, but don't flame me for not wanting to maintain manually added backwards compatibility CMake code for a theoretical existing feature nobody asked for.

Reply:
I don't see where I'm "flaming you", but rather the opposite: "pure madness", "random ... decision", "work of the devils", "I don't like that", ... mixed with claims for which I brought arguments that they are wrong ("(Boost)MPI is not required")   
I brought arguments for why the MPI dependency should be included in the libraries CMake and why it is beneficial to use a method, that works in more versions of CMake.   
I provided code that achieves this. It is easy, short, understandable and (pretty much) the standard way for CMake modules, not defining targets. I don't see where this hurts anyone: No need to change this, no need to write it (as I already did).

It seems we have different goals. I want this library to be as compatible as reasonably possible so it can be widely used. This means keeping requirements low. And if one can avoid raising a requirement by ~5 lines of trivial code, I deem that reasonable. The alternative is less people being able to use this. And if one goes down that path, where do you stop? "We can save 2 lines, when using C++20", "Another 3 when using CMake 3.12", "Oh hey, Boost 1.67 has this cool thing!", ... I just ask: Is this worth it? What does it cost you to have those 5 lines?

However I already offered to replace this by a simple error message. Currently this CMake requires 3.8 but uses a 3.9 feature and nowhere mentions this. So either:
- Raise the minimum (which I'd ask to not, as e.g. we won't be able to use this)
- Copy in the FindMPI from a newer CMake (ew...)
- Use this simple code to allow compatibility with 3.8+
- Add a message or at least docu